bftengine/src/bftengine/DebugStatistics.cpp

```diff
@@ -1,205 +1,160 @@
-//Concord
+// Concord
 //
-//Copyright (c) 2018 VMware, Inc. All Rights Reserved.
+// Copyright (c) 2018 VMware, Inc. All Rights Reserved.
 //
-//This product is licensed to you under the Apache 2.0 license (the "License").  You may not use this product except in compliance with the Apache 2.0 License. 
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0 License.
 //
-//This product may include a number of subcomponents with separate copyright notices and license terms. Your use of these subcomponents is subject to the terms and conditions of the subcomponent's license, as noted in the LICENSE file.
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
 
 #include "DebugStatistics.hpp"
 
-
 #ifdef DEBUG_STATISTICS
 
 #if defined(_WIN32)
 #include <winsock2.h>
 #include <WinBase.h>
-#else
-#include <sys/time.h>
 #endif
 #include <inttypes.h>
 #include <stdio.h>
 #include "MsgCode.hpp"
+#include "Logger.hpp"
+
+using namespace std;
+
+namespace bftEngine {
+namespace impl {
+
+void DebugStatistics::onCycleCheck() {
+  DebugStatDesc& d = getDebugStatDesc();
+
+  if (!d.initialized)  // if this is the first time
+  {
+    d.initialized = true;
+    d.lastCycleTime = getMonotonicTime();
+    clearCounters(d);
+    return;
+  }
+
+  Time currTime = getMonotonicTime();
+
+  long long durationMicroSec = subtract(currTime, d.lastCycleTime);
+
+  double durationSec = durationMicroSec / 1000000.0;
+
+  if (durationSec < DEBUG_STAT_PERIOD_SECONDS) return;
+
+  double readThruput = 0;
+  double writeThruput = 0;
+
+  if (d.completedReadOnlyRequests > 0)
+    readThruput = (d.completedReadOnlyRequests / durationSec);
+
+  if (d.completedReadWriteRequests > 0)
+    writeThruput = (d.completedReadWriteRequests / durationSec);
+
+  // We use INFO logging instead of DEBUG logging since there is already a
+  // separate switch to turn DebugStatistics on and off. It's likely we want
+  // this data to appear when we do not want full debug logging.
+  LOG_INFO(GL, " STAT:" << endl <<
+      "    ReadOnlyThroughput = " << fixed << setprecision(2) 
```

Comment:
```
concord-bft/bftengine/src/bftengine/DebugStatistics.cpp:63:47: error: use of undeclared identifier 'setprecision'
      "    ReadOnlyThroughput = " << fixed << setprecision(2) 
```

I think you need to `#include <iomanip>`.

Reply:
WTF. I built this using both clang and gcc on Ubuntu 18.04 and neither complained. I'm really confused here.

Reply:
I can certainly add `#include <iomanip>`, but I'd like to understand why my system doesn't seem to need it.

Reply:
Ok, I just went ahead and added the header. I have no idea why it wasn't needed on my system. I also removed the windows related includes, and stdio.h since they aren't needed.